output/cloud/expv2/output.go

```diff
@@ -19,8 +21,16 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// requestMetadatasCollector is an interface for collecting request metadatas
+// and retrieving them, so they can be flushed using a flusher.
+type requestMetadatasCollector interface {
+	CollectRequestMetadatas([]metrics.SampleContainer)
+	PopAll() insights.RequestMetadatas
+}
+
+// flusher is an interface for flushing data to the cloud.
 type flusher interface {
-	flush(context.Context) error
+	flush(ctx context.Context) error
```

Comment:
```suggestion
	flush(context.Context) error
```
I don't think we need it the type already documents it clearly.